src/Container/ApplicationConfigInjectionDelegator.php

```diff
@@ -1,19 +1,49 @@
 <?php
 /**
  * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
- * @copyright Copyright (c) 2016-2017 Zend Technologies USA Inc. (https://www.zend.com)
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
  * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
  */
 
 declare(strict_types=1);
 
-namespace Zend\Expressive;
+namespace Zend\Expressive\Container;
 
+use Psr\Container\ContainerInterface;
 use SplPriorityQueue;
+use Zend\Expressive\Application;
+use Zend\Expressive\Exception\InvalidArgumentException;
+use Zend\Expressive\Middleware;
 use Zend\Expressive\Router\Route;
 
-trait ApplicationConfigInjectionTrait
+class ApplicationConfigInjectionDelegator
 {
+    /**
+     * Decorate an Application instance by injecting routes and/or middleware
+     * from configuration.
+     */
+    public function __invoke(ContainerInterface $container, string $serviceName, callable $callback) : Application
+    {
+        $application = $callback();
+        if (! $application instanceof Application) {
+            return $application;
```

Comment:
It looks like we will have here TypeError, because method declaration has `Application` return type and we return here `$application` which is not instance of `Application`.

Reply:
Good point; I'll remove the typehint from the method.

Reply:
Done!